flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/TimerSerializer.java

```diff
@@ -208,8 +215,35 @@ public TypeSerializerConfigSnapshot snapshotConfiguration() {
 	@Override
 	public CompatibilityResult<TimerHeapInternalTimer<K, N>> ensureCompatibility(
 		TypeSerializerConfigSnapshot configSnapshot) {
-		//TODO this is just a mock (assuming no serializer updates) for now and needs a proper implementation! change this before release.
-		return CompatibilityResult.compatible();
+
+		if (configSnapshot instanceof CompositeTypeSerializerConfigSnapshot) {
```

Comment:
should this actually be `configSnapshot instanceof TimerSerializerConfigSnapshot`?